include/revng/Pipeline/AnyDiff.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <any>
+
+#include "revng/Pipeline/InvalidationEvent.h"
+#include "revng/Pipeline/TupleTreeInvalidationEvent.h"
+#include "revng/Support/YAMLTraits.h"
+#include "revng/TupleTree/TupleTreeDiff.h"
+
+namespace pipeline {
+class AnyDiff {
+public:
+  using SerializationMethod = void (*)(llvm::raw_ostream &OS,
+                                       const AnyDiff &Diff);
+
+  template<TupleTreeCompatible T>
+  static void serializeImpl(llvm::raw_ostream &OS, const AnyDiff &Diff) {
+    auto &Casted = Diff.getAs<T>();
+
+    ::serialize(OS, Casted);
+  }
+
+private:
+  std::unique_ptr<InvalidationEventBase> Event;
+  std::any Diff;
+  SerializationMethod Serialize;
```

Comment:
This is weird. `std::any` is type erasing the destructor and then you have `Serialize`.
Wouldn't it be better not to use `std::any` and just have a `virtual` class with one inheritor for each class to type erase (i.e., classical type-erasure)?